### DISCUSSION TITLE
FPP Annotations Bug

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1778] [Q&A] [answered]
Hi folks,

Before I go and open a bug report on the FPP repo, I wanted to see if anyone else has had this problem.

I this (among many examples) in the FPP file of my component:
```
        # ......................................................................
        @ Commands the Some Application to send its jabberwocky state-machine
        @ state log as telemetry.
        guarded command CMD_READ_JABBERWOCKY_SM_STATE_LOG \
        opcode 0
```

During the call of `fprime-util impl`, FPP generates the following broken code in the template header:
```cpp
      //! Implementation for CMD_READ_JABBERWOCKY_SM_STATE_LOG command handler
      //! Commands the Some Application to send its jabberwocky state-machine
      state log as telemetry.
      void CMD_READ_JABBERWOCKY_SM_STATE_LOG_cmdHandler(
          const FwOpcodeType opCode, /*!< The opcode*/
          const U32 cmdSeq /*!< The command sequence number*/
      );
```

Has anyone experienced a similar problem and been able to fix it? My first reaction was to update the `fprime-fpp` pip package, but then `fprime-utils` complains.

**Comment by timcanham** [comment]
Is the issue that the second annotation line has no comment character in the generated code?

**Reply by JackNWhite** [reply]
Yes - I should have said that I am on FPP v1.02.

**Comment by LeStarch** [comment, marked as answer]
This has been fixed in later versions of FPP (as near as I can tell).  You have several options:

1. Use single-line annotations
2. Upgrade `fprime` and with it fpp
3. Upgrade fpp only

Option 1 is the safest.  F´  ships with a specific set of supported tools versions.  Thus keeping your version of F´  and changing how you write FPP files will have the least overall impact.

Option 2 will require some work. As F´  develops, some patterns of usage change. You will need to update F´and your deployment to follow the new patterns.

Option 3 may or may not work at all.  FPP and F´  have requirements on each other, so bumping the FPP version beyond what F´ specifies may simply not work as it is not the required version.  However, if you want to try it, change `fprime/requirements.txt` to list your desired version of FPP and build it.  If it builds, then you should be good.

**Reply by bocchino** [reply]
See below. This is a Python autocoder glitch, not an FPP version issue.

**Comment by bocchino** [comment]
This looks to me like a bug in the XML-to-C++ autocoder for component implementations. FPP currently has no responsibility for generating those component skeletons. It just generates XML. If there were an FPP bug, the XML would be mangled, and it wouldn't make it through the Python parser.

**Reply by bocchino** [reply]
I confirmed this is an idiosyncrasy in the Python impl generator. It's still there:

https://github.com/nasa/fprime/blob/ee44a890b39fc58bdaa5e29772f7a94f766e83c2/Autocoders/Python/src/fprime_ac/generators/templates/impl/hpp.tmpl#L88

Multiline comments aren't supported properly. The workaround for now is to fix up the comments by hand. This is impl code that you have to hand-modify anyway. Changing the FPP version will not have any effect.
